src/fft.jl

```diff
@@ -207,8 +207,8 @@ generic_idct(a::AbstractArray{T}) where {T <: AbstractFloat} = real(generic_idct
 for f in (:dct, :dct!, :idct, :idct!)
     pf = Symbol("plan_", f)
     @eval begin
-        $f(x::AbstractArray{<:AbstractFloats}) = $pf(x) * x
-        $f(x::AbstractArray{<:AbstractFloats}, region) = $pf(x, region) * x
+        $f(x::AbstractArray{<:AbstractFloats}) = $pf(x, size(x)) * x
+        $f(x::AbstractArray{<:AbstractFloats}, region) = $pf(x, size(x), region) * x
```

Comment:
I would try to avoid changing the interface of the plan_X functions. Here, you're adding size(x) as a second argument, but AbstractFFTS adds the region as a second argument [here](https://github.com/JuliaMath/AbstractFFTs.jl/blob/0918c3c36424865854bc498a5f6b78a9bb09bece/src/definitions.jl#L64). It is the construct of the plan object that takes X as an argument and then computes its size, see e.g. [here](https://github.com/JuliaMath/FFTW.jl/blob/17bc81a0fcf9875d777ea4bee2fca70fc23c8a0c/src/fft.jl#L265). It seems safer to mimick that?

Reply:
Ah, so different sizes are due to complex-real differences. Ok.